src/vl-map-features-layer.js

```diff
@@ -0,0 +1,186 @@
+ * @property {number} data-vl-auto-extent-max-zoom - Attribuut geeft aan tot op welk niveau er maximaal automatisch gezoomd wordt bij een extent.
+ * @property {boolean} data-vl-cluster - Attribuut geeft aan of de features geclusterd moeten worden of niet.
+ * @property {number} data-vl-cluster-distance - Attribuut geeft aan vanaf welke afstand tussen features er geclusterd mag worden.
+ * @property {string[]} data-vl-features - Attribuut die de kaartlaag bevat.
+ *
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-map-features-layer.html|Demo}
+ */
+export class VlMapFeaturesLayer extends VlMapVectorLayer {
+  static get _observedAttributes() {
+    return ['auto-extent', 'features'];
+  }
+
+  constructor() {
+    super();
+    this._geoJSON = new OlGeoJSON();
+    this._source = this.__createSource();
+    this._layer = this._createLayer();
+  }
+
+  async connectedCallback() {
+    await super.connectedCallback();
+    if (this.mapElement) {
+      this._autoZoomToExtent();
+    }
+  }
+
+  /**
+   * Geeft de OpenLayers features collectie van de kaartlaag terug.
+   *
+   * @return {object}
+   */
+  get features() {
+    const features = this.getAttribute('features');
+    return features ? this._geoJSON.readFeatures(features) : [];
+  }
+
+  /**
+   * Zet de OpenLayers features collectie op de kaartlaag.
+   *
+   * @param {object} features
+   */
+  set features(features) {
+    this.setAttribute('features', JSON.stringify(features));
+  }
+
+  get _autoExtent() {
+    return this.getAttribute('auto-extent') != undefined;
+  }
+
+  get _autoExtentMaxZoom() {
+    return this.getAttribute('auto-extent-max-zoom');
+  }
+
+  get cluster() {
+    return this.getAttribute('cluster') != undefined;
+  }
+
+  get _clusterDistance() {
+    return this.getAttribute('cluster-distance');
+  }
+
+  /**
+   * Verwijdert de stijl van al de kaartlaag features.
+   */
+  removeFeaturesStyle() {
+    if (this.source && this.source.getFeatures()) {
+      this.source.getFeatures().forEach((feature) => {
+        feature.setStyle(null);
+      });
+    }
+  }
+
+  /**
+   * Geeft de feature terug op basis van het id attribuut.
+   *
+   * @param {number} id
+   * @return {Object}
+   */
+  getFeature(id) {
+    if (this.source && this.source.getFeatures()) {
+      return this.source.getFeatures().filter((feature) => {
+        return feature.getId() === id;
+      })[0];
+    }
+  }
+
+  /**
+   * Geeft de cluster terug op basis van het id attribuut.
+   *
+   * @param {number} id
+   * @return {boolean}
+   */
+  getCluster(id) {
+    if (this._layer) {
+      return this._layer.getSource().getFeatures().filter((cluster) => {
+        const features = cluster.get('features');
+        if (features) {
+          return features.some((feature) => {
```

Comment:
uit interesse, de some functie waar wordt die geimplementeerd. Zag ze niet staan in de ol feature dus het is wrs iets uit onze code.

Reply:
Ik geloof dat cluster.get('features') gewoon een array gaat teruggeven, en een javascript array biedt `some()` aan: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some
zelfde als anyMatch dus

Reply:
nice